// @azure/communication-call-automation/dist/react-native/generated/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CommunicationIdentifierModelKind} that the service accepts. */
export var KnownCommunicationIdentifierModelKind;
(function (KnownCommunicationIdentifierModelKind) {
    /** Unknown */
    KnownCommunicationIdentifierModelKind["Unknown"] = "unknown";
    /** CommunicationUser */
    KnownCommunicationIdentifierModelKind["CommunicationUser"] = "communicationUser";
    /** PhoneNumber */
    KnownCommunicationIdentifierModelKind["PhoneNumber"] = "phoneNumber";
    /** MicrosoftTeamsUser */
    KnownCommunicationIdentifierModelKind["MicrosoftTeamsUser"] = "microsoftTeamsUser";
    /** MicrosoftTeamsApp */
    KnownCommunicationIdentifierModelKind["MicrosoftTeamsApp"] = "microsoftTeamsApp";
})(KnownCommunicationIdentifierModelKind || (KnownCommunicationIdentifierModelKind = {}));
/** Known values of {@link CommunicationCloudEnvironmentModel} that the service accepts. */
export var KnownCommunicationCloudEnvironmentModel;
(function (KnownCommunicationCloudEnvironmentModel) {
    /** Public */
    KnownCommunicationCloudEnvironmentModel["Public"] = "public";
    /** Dod */
    KnownCommunicationCloudEnvironmentModel["Dod"] = "dod";
    /** Gcch */
    KnownCommunicationCloudEnvironmentModel["Gcch"] = "gcch";
})(KnownCommunicationCloudEnvironmentModel || (KnownCommunicationCloudEnvironmentModel = {}));
/** Known values of {@link CallConnectionStateModel} that the service accepts. */
export var KnownCallConnectionStateModel;
(function (KnownCallConnectionStateModel) {
    /** Unknown */
    KnownCallConnectionStateModel["Unknown"] = "unknown";
    /** Connecting */
    KnownCallConnectionStateModel["Connecting"] = "connecting";
    /** Connected */
    KnownCallConnectionStateModel["Connected"] = "connected";
    /** Transferring */
    KnownCallConnectionStateModel["Transferring"] = "transferring";
    /** TransferAccepted */
    KnownCallConnectionStateModel["TransferAccepted"] = "transferAccepted";
    /** Disconnecting */
    KnownCallConnectionStateModel["Disconnecting"] = "disconnecting";
    /** Disconnected */
    KnownCallConnectionStateModel["Disconnected"] = "disconnected";
})(KnownCallConnectionStateModel || (KnownCallConnectionStateModel = {}));
/** Known values of {@link CallRejectReason} that the service accepts. */
export var KnownCallRejectReason;
(function (KnownCallRejectReason) {
    /** None */
    KnownCallRejectReason["None"] = "none";
    /** Busy */
    KnownCallRejectReason["Busy"] = "busy";
    /** Forbidden */
    KnownCallRejectReason["Forbidden"] = "forbidden";
})(KnownCallRejectReason || (KnownCallRejectReason = {}));
/** Known values of {@link CallLocatorKind} that the service accepts. */
export var KnownCallLocatorKind;
(function (KnownCallLocatorKind) {
    /** Unknown */
    KnownCallLocatorKind["Unknown"] = "unknown";
    /** GroupCallLocator */
    KnownCallLocatorKind["GroupCallLocator"] = "groupCallLocator";
    /** ServerCallLocator */
    KnownCallLocatorKind["ServerCallLocator"] = "serverCallLocator";
    /** RoomCallLocator */
    KnownCallLocatorKind["RoomCallLocator"] = "roomCallLocator";
})(KnownCallLocatorKind || (KnownCallLocatorKind = {}));
/** Known values of {@link PlaySourceType} that the service accepts. */
export var KnownPlaySourceType;
(function (KnownPlaySourceType) {
    /** File */
    KnownPlaySourceType["File"] = "file";
    /** Text */
    KnownPlaySourceType["Text"] = "text";
    /** Ssml */
    KnownPlaySourceType["Ssml"] = "ssml";
})(KnownPlaySourceType || (KnownPlaySourceType = {}));
/** Known values of {@link VoiceKind} that the service accepts. */
export var KnownVoiceKind;
(function (KnownVoiceKind) {
    /** Male */
    KnownVoiceKind["Male"] = "male";
    /** Female */
    KnownVoiceKind["Female"] = "female";
})(KnownVoiceKind || (KnownVoiceKind = {}));
/** Known values of {@link RecognizeInputType} that the service accepts. */
export var KnownRecognizeInputType;
(function (KnownRecognizeInputType) {
    /** Dtmf */
    KnownRecognizeInputType["Dtmf"] = "dtmf";
    /** Speech */
    KnownRecognizeInputType["Speech"] = "speech";
    /** SpeechOrDtmf */
    KnownRecognizeInputType["SpeechOrDtmf"] = "speechOrDtmf";
    /** Choices */
    KnownRecognizeInputType["Choices"] = "choices";
})(KnownRecognizeInputType || (KnownRecognizeInputType = {}));
/** Known values of {@link Tone} that the service accepts. */
export var KnownTone;
(function (KnownTone) {
    /** Zero */
    KnownTone["Zero"] = "zero";
    /** One */
    KnownTone["One"] = "one";
    /** Two */
    KnownTone["Two"] = "two";
    /** Three */
    KnownTone["Three"] = "three";
    /** Four */
    KnownTone["Four"] = "four";
    /** Five */
    KnownTone["Five"] = "five";
    /** Six */
    KnownTone["Six"] = "six";
    /** Seven */
    KnownTone["Seven"] = "seven";
    /** Eight */
    KnownTone["Eight"] = "eight";
    /** Nine */
    KnownTone["Nine"] = "nine";
    /** A */
    KnownTone["A"] = "a";
    /** B */
    KnownTone["B"] = "b";
    /** C */
    KnownTone["C"] = "c";
    /** D */
    KnownTone["D"] = "d";
    /** Pound */
    KnownTone["Pound"] = "pound";
    /** Asterisk */
    KnownTone["Asterisk"] = "asterisk";
})(KnownTone || (KnownTone = {}));
/** Known values of {@link RecordingContentType} that the service accepts. */
export var KnownRecordingContentType;
(function (KnownRecordingContentType) {
    /** Audio */
    KnownRecordingContentType["Audio"] = "audio";
    /** AudioVideo */
    KnownRecordingContentType["AudioVideo"] = "audioVideo";
})(KnownRecordingContentType || (KnownRecordingContentType = {}));
/** Known values of {@link RecordingChannelType} that the service accepts. */
export var KnownRecordingChannelType;
(function (KnownRecordingChannelType) {
    /** Mixed */
    KnownRecordingChannelType["Mixed"] = "mixed";
    /** Unmixed */
    KnownRecordingChannelType["Unmixed"] = "unmixed";
})(KnownRecordingChannelType || (KnownRecordingChannelType = {}));
/** Known values of {@link RecordingFormatType} that the service accepts. */
export var KnownRecordingFormatType;
(function (KnownRecordingFormatType) {
    /** Wav */
    KnownRecordingFormatType["Wav"] = "wav";
    /** Mp3 */
    KnownRecordingFormatType["Mp3"] = "mp3";
    /** Mp4 */
    KnownRecordingFormatType["Mp4"] = "mp4";
})(KnownRecordingFormatType || (KnownRecordingFormatType = {}));
/** Known values of {@link RecordingStorageKind} that the service accepts. */
export var KnownRecordingStorageKind;
(function (KnownRecordingStorageKind) {
    /** Storage managed by Azure Communication Services */
    KnownRecordingStorageKind["AzureCommunicationServices"] = "AzureCommunicationServices";
    /** Storage managed by provided Azure blob */
    KnownRecordingStorageKind["AzureBlobStorage"] = "AzureBlobStorage";
})(KnownRecordingStorageKind || (KnownRecordingStorageKind = {}));
/** Known values of {@link RecordingState} that the service accepts. */
export var KnownRecordingState;
(function (KnownRecordingState) {
    /** Active */
    KnownRecordingState["Active"] = "active";
    /** Inactive */
    KnownRecordingState["Inactive"] = "inactive";
})(KnownRecordingState || (KnownRecordingState = {}));
/** Known values of {@link RecordingKind} that the service accepts. */
export var KnownRecordingKind;
(function (KnownRecordingKind) {
    /** Recording initiated by Azure Communication Services */
    KnownRecordingKind["AzureCommunicationServices"] = "AzureCommunicationServices";
    /** Recording initiated by Teams user */
    KnownRecordingKind["Teams"] = "Teams";
    /** Recording initiated by Teams compliance policy */
    KnownRecordingKind["TeamsCompliance"] = "TeamsCompliance";
})(KnownRecordingKind || (KnownRecordingKind = {}));
/** Known values of {@link RecognitionType} that the service accepts. */
export var KnownRecognitionType;
(function (KnownRecognitionType) {
    /** Dtmf */
    KnownRecognitionType["Dtmf"] = "dtmf";
    /** Speech */
    KnownRecognitionType["Speech"] = "speech";
    /** Choices */
    KnownRecognitionType["Choices"] = "choices";
})(KnownRecognitionType || (KnownRecognitionType = {}));
