// @azure/communication-call-automation/dist/esm/generated/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const CreateCallRequest = {
    type: {
        name: "Composite",
        className: "CreateCallRequest",
        modelProperties: {
            targets: {
                serializedName: "targets",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommunicationIdentifierModel",
                        },
                    },
                },
            },
            sourceCallerIdNumber: {
                serializedName: "sourceCallerIdNumber",
                type: {
                    name: "Composite",
                    className: "PhoneNumberIdentifierModel",
                },
            },
            sourceDisplayName: {
                serializedName: "sourceDisplayName",
                type: {
                    name: "String",
                },
            },
            source: {
                serializedName: "source",
                type: {
                    name: "Composite",
                    className: "CommunicationUserIdentifierModel",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            callbackUri: {
                serializedName: "callbackUri",
                required: true,
                type: {
                    name: "String",
                },
            },
            callIntelligenceOptions: {
                serializedName: "callIntelligenceOptions",
                type: {
                    name: "Composite",
                    className: "CallIntelligenceOptionsInternal",
                },
            },
        },
    },
};
export const CommunicationIdentifierModel = {
    type: {
        name: "Composite",
        className: "CommunicationIdentifierModel",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            rawId: {
                serializedName: "rawId",
                type: {
                    name: "String",
                },
            },
            communicationUser: {
                serializedName: "communicationUser",
                type: {
                    name: "Composite",
                    className: "CommunicationUserIdentifierModel",
                },
            },
            phoneNumber: {
                serializedName: "phoneNumber",
                type: {
                    name: "Composite",
                    className: "PhoneNumberIdentifierModel",
                },
            },
            microsoftTeamsUser: {
                serializedName: "microsoftTeamsUser",
                type: {
                    name: "Composite",
                    className: "MicrosoftTeamsUserIdentifierModel",
                },
            },
            microsoftTeamsApp: {
                serializedName: "microsoftTeamsApp",
                type: {
                    name: "Composite",
                    className: "MicrosoftTeamsAppIdentifierModel",
                },
            },
        },
    },
};
export const CommunicationUserIdentifierModel = {
    type: {
        name: "Composite",
        className: "CommunicationUserIdentifierModel",
        modelProperties: {
            id: {
                serializedName: "id",
                required: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const PhoneNumberIdentifierModel = {
    type: {
        name: "Composite",
        className: "PhoneNumberIdentifierModel",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const MicrosoftTeamsUserIdentifierModel = {
    type: {
        name: "Composite",
        className: "MicrosoftTeamsUserIdentifierModel",
        modelProperties: {
            userId: {
                serializedName: "userId",
                required: true,
                type: {
                    name: "String",
                },
            },
            isAnonymous: {
                serializedName: "isAnonymous",
                type: {
                    name: "Boolean",
                },
            },
            cloud: {
                serializedName: "cloud",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const MicrosoftTeamsAppIdentifierModel = {
    type: {
        name: "Composite",
        className: "MicrosoftTeamsAppIdentifierModel",
        modelProperties: {
            appId: {
                serializedName: "appId",
                required: true,
                type: {
                    name: "String",
                },
            },
            cloud: {
                serializedName: "cloud",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const CallIntelligenceOptionsInternal = {
    type: {
        name: "Composite",
        className: "CallIntelligenceOptionsInternal",
        modelProperties: {
            cognitiveServicesEndpoint: {
                serializedName: "cognitiveServicesEndpoint",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const CallConnectionPropertiesInternal = {
    type: {
        name: "Composite",
        className: "CallConnectionPropertiesInternal",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            targets: {
                serializedName: "targets",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommunicationIdentifierModel",
                        },
                    },
                },
            },
            callConnectionState: {
                serializedName: "callConnectionState",
                type: {
                    name: "String",
                },
            },
            callbackUri: {
                serializedName: "callbackUri",
                type: {
                    name: "String",
                },
            },
            sourceCallerIdNumber: {
                serializedName: "sourceCallerIdNumber",
                type: {
                    name: "Composite",
                    className: "PhoneNumberIdentifierModel",
                },
            },
            sourceDisplayName: {
                serializedName: "sourceDisplayName",
                type: {
                    name: "String",
                },
            },
            source: {
                serializedName: "source",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            answeredBy: {
                serializedName: "answeredBy",
                type: {
                    name: "Composite",
                    className: "CommunicationUserIdentifierModel",
                },
            },
            answeredFor: {
                serializedName: "answeredFor",
                type: {
                    name: "Composite",
                    className: "PhoneNumberIdentifierModel",
                },
            },
        },
    },
};
export const CommunicationErrorResponse = {
    type: {
        name: "Composite",
        className: "CommunicationErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "CommunicationError",
                },
            },
        },
    },
};
export const CommunicationError = {
    type: {
        name: "Composite",
        className: "CommunicationError",
        modelProperties: {
            code: {
                serializedName: "code",
                required: true,
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                required: true,
                type: {
                    name: "String",
                },
            },
            target: {
                serializedName: "target",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommunicationError",
                        },
                    },
                },
            },
            innerError: {
                serializedName: "innererror",
                type: {
                    name: "Composite",
                    className: "CommunicationError",
                },
            },
        },
    },
};
export const AnswerCallRequest = {
    type: {
        name: "Composite",
        className: "AnswerCallRequest",
        modelProperties: {
            incomingCallContext: {
                serializedName: "incomingCallContext",
                required: true,
                type: {
                    name: "String",
                },
            },
            callbackUri: {
                serializedName: "callbackUri",
                required: true,
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            callIntelligenceOptions: {
                serializedName: "callIntelligenceOptions",
                type: {
                    name: "Composite",
                    className: "CallIntelligenceOptionsInternal",
                },
            },
            answeredBy: {
                serializedName: "answeredBy",
                type: {
                    name: "Composite",
                    className: "CommunicationUserIdentifierModel",
                },
            },
        },
    },
};
export const RedirectCallRequest = {
    type: {
        name: "Composite",
        className: "RedirectCallRequest",
        modelProperties: {
            incomingCallContext: {
                serializedName: "incomingCallContext",
                required: true,
                type: {
                    name: "String",
                },
            },
            target: {
                serializedName: "target",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
        },
    },
};
export const RejectCallRequest = {
    type: {
        name: "Composite",
        className: "RejectCallRequest",
        modelProperties: {
            incomingCallContext: {
                serializedName: "incomingCallContext",
                required: true,
                type: {
                    name: "String",
                },
            },
            callRejectReason: {
                serializedName: "callRejectReason",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const ConnectRequest = {
    type: {
        name: "Composite",
        className: "ConnectRequest",
        modelProperties: {
            callLocator: {
                serializedName: "callLocator",
                type: {
                    name: "Composite",
                    className: "CallLocator",
                },
            },
            callbackUri: {
                serializedName: "callbackUri",
                required: true,
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            callIntelligenceOptions: {
                serializedName: "callIntelligenceOptions",
                type: {
                    name: "Composite",
                    className: "CallIntelligenceOptionsInternal",
                },
            },
        },
    },
};
export const CallLocator = {
    type: {
        name: "Composite",
        className: "CallLocator",
        modelProperties: {
            groupCallId: {
                serializedName: "groupCallId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            roomId: {
                serializedName: "roomId",
                type: {
                    name: "String",
                },
            },
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const TransferToParticipantRequest = {
    type: {
        name: "Composite",
        className: "TransferToParticipantRequest",
        modelProperties: {
            targetParticipant: {
                serializedName: "targetParticipant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            transferee: {
                serializedName: "transferee",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
            customCallingContext: {
                serializedName: "customCallingContext",
                type: {
                    name: "Composite",
                    className: "CustomCallingContextInternal",
                },
            },
            sourceCallerIdNumber: {
                serializedName: "sourceCallerIdNumber",
                type: {
                    name: "Composite",
                    className: "PhoneNumberIdentifierModel",
                },
            },
        },
    },
};
export const CustomCallingContextInternal = {
    type: {
        name: "Composite",
        className: "CustomCallingContextInternal",
        modelProperties: {
            voipHeaders: {
                serializedName: "voipHeaders",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
            sipHeaders: {
                serializedName: "sipHeaders",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
export const TransferCallResponse = {
    type: {
        name: "Composite",
        className: "TransferCallResponse",
        modelProperties: {
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const PlayRequest = {
    type: {
        name: "Composite",
        className: "PlayRequest",
        modelProperties: {
            playSources: {
                serializedName: "playSources",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PlaySourceInternal",
                        },
                    },
                },
            },
            playTo: {
                serializedName: "playTo",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommunicationIdentifierModel",
                        },
                    },
                },
            },
            interruptCallMediaOperation: {
                serializedName: "interruptCallMediaOperation",
                type: {
                    name: "Boolean",
                },
            },
            playOptions: {
                serializedName: "playOptions",
                type: {
                    name: "Composite",
                    className: "PlayOptionsInternal",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const PlaySourceInternal = {
    type: {
        name: "Composite",
        className: "PlaySourceInternal",
        modelProperties: {
            kind: {
                serializedName: "kind",
                required: true,
                type: {
                    name: "String",
                },
            },
            playSourceCacheId: {
                serializedName: "playSourceCacheId",
                type: {
                    name: "String",
                },
            },
            file: {
                serializedName: "file",
                type: {
                    name: "Composite",
                    className: "FileSourceInternal",
                },
            },
            text: {
                serializedName: "text",
                type: {
                    name: "Composite",
                    className: "TextSourceInternal",
                },
            },
            ssml: {
                serializedName: "ssml",
                type: {
                    name: "Composite",
                    className: "SsmlSourceInternal",
                },
            },
        },
    },
};
export const FileSourceInternal = {
    type: {
        name: "Composite",
        className: "FileSourceInternal",
        modelProperties: {
            uri: {
                serializedName: "uri",
                required: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const TextSourceInternal = {
    type: {
        name: "Composite",
        className: "TextSourceInternal",
        modelProperties: {
            text: {
                serializedName: "text",
                required: true,
                type: {
                    name: "String",
                },
            },
            sourceLocale: {
                serializedName: "sourceLocale",
                type: {
                    name: "String",
                },
            },
            voiceKind: {
                serializedName: "voiceKind",
                type: {
                    name: "String",
                },
            },
            voiceName: {
                serializedName: "voiceName",
                type: {
                    name: "String",
                },
            },
            customVoiceEndpointId: {
                serializedName: "customVoiceEndpointId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const SsmlSourceInternal = {
    type: {
        name: "Composite",
        className: "SsmlSourceInternal",
        modelProperties: {
            ssmlText: {
                serializedName: "ssmlText",
                required: true,
                type: {
                    name: "String",
                },
            },
            customVoiceEndpointId: {
                serializedName: "customVoiceEndpointId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const PlayOptionsInternal = {
    type: {
        name: "Composite",
        className: "PlayOptionsInternal",
        modelProperties: {
            loop: {
                serializedName: "loop",
                required: true,
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
export const RecognizeRequest = {
    type: {
        name: "Composite",
        className: "RecognizeRequest",
        modelProperties: {
            recognizeInputType: {
                serializedName: "recognizeInputType",
                required: true,
                type: {
                    name: "String",
                },
            },
            playPrompt: {
                serializedName: "playPrompt",
                type: {
                    name: "Composite",
                    className: "PlaySourceInternal",
                },
            },
            playPrompts: {
                serializedName: "playPrompts",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PlaySourceInternal",
                        },
                    },
                },
            },
            interruptCallMediaOperation: {
                serializedName: "interruptCallMediaOperation",
                type: {
                    name: "Boolean",
                },
            },
            recognizeOptions: {
                serializedName: "recognizeOptions",
                type: {
                    name: "Composite",
                    className: "RecognizeOptions",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const RecognizeOptions = {
    type: {
        name: "Composite",
        className: "RecognizeOptions",
        modelProperties: {
            interruptPrompt: {
                serializedName: "interruptPrompt",
                type: {
                    name: "Boolean",
                },
            },
            initialSilenceTimeoutInSeconds: {
                constraints: {
                    InclusiveMaximum: 300,
                    InclusiveMinimum: 0,
                },
                serializedName: "initialSilenceTimeoutInSeconds",
                type: {
                    name: "Number",
                },
            },
            targetParticipant: {
                serializedName: "targetParticipant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            speechLanguage: {
                serializedName: "speechLanguage",
                type: {
                    name: "String",
                },
            },
            speechRecognitionModelEndpointId: {
                serializedName: "speechRecognitionModelEndpointId",
                type: {
                    name: "String",
                },
            },
            dtmfOptions: {
                serializedName: "dtmfOptions",
                type: {
                    name: "Composite",
                    className: "DtmfOptions",
                },
            },
            choices: {
                serializedName: "choices",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Choice",
                        },
                    },
                },
            },
            speechOptions: {
                serializedName: "speechOptions",
                type: {
                    name: "Composite",
                    className: "SpeechOptions",
                },
            },
        },
    },
};
export const DtmfOptions = {
    type: {
        name: "Composite",
        className: "DtmfOptions",
        modelProperties: {
            interToneTimeoutInSeconds: {
                constraints: {
                    InclusiveMaximum: 60,
                    InclusiveMinimum: 1,
                },
                serializedName: "interToneTimeoutInSeconds",
                type: {
                    name: "Number",
                },
            },
            maxTonesToCollect: {
                serializedName: "maxTonesToCollect",
                type: {
                    name: "Number",
                },
            },
            stopTones: {
                serializedName: "stopTones",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
export const Choice = {
    type: {
        name: "Composite",
        className: "Choice",
        modelProperties: {
            label: {
                serializedName: "label",
                required: true,
                type: {
                    name: "String",
                },
            },
            phrases: {
                serializedName: "phrases",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            tone: {
                serializedName: "tone",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const SpeechOptions = {
    type: {
        name: "Composite",
        className: "SpeechOptions",
        modelProperties: {
            endSilenceTimeoutInMs: {
                serializedName: "endSilenceTimeoutInMs",
                type: {
                    name: "Number",
                },
            },
        },
    },
};
export const ContinuousDtmfRecognitionRequest = {
    type: {
        name: "Composite",
        className: "ContinuousDtmfRecognitionRequest",
        modelProperties: {
            targetParticipant: {
                serializedName: "targetParticipant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const SendDtmfTonesRequest = {
    type: {
        name: "Composite",
        className: "SendDtmfTonesRequest",
        modelProperties: {
            tones: {
                serializedName: "tones",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            targetParticipant: {
                serializedName: "targetParticipant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const SendDtmfTonesResult = {
    type: {
        name: "Composite",
        className: "SendDtmfTonesResult",
        modelProperties: {
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const HoldRequest = {
    type: {
        name: "Composite",
        className: "HoldRequest",
        modelProperties: {
            targetParticipant: {
                serializedName: "targetParticipant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            playSourceInfo: {
                serializedName: "playSourceInfo",
                type: {
                    name: "Composite",
                    className: "PlaySourceInternal",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const UnholdRequest = {
    type: {
        name: "Composite",
        className: "UnholdRequest",
        modelProperties: {
            targetParticipant: {
                serializedName: "targetParticipant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const GetParticipantsResponse = {
    type: {
        name: "Composite",
        className: "GetParticipantsResponse",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CallParticipantInternal",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const CallParticipantInternal = {
    type: {
        name: "Composite",
        className: "CallParticipantInternal",
        modelProperties: {
            identifier: {
                serializedName: "identifier",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            isMuted: {
                serializedName: "isMuted",
                type: {
                    name: "Boolean",
                },
            },
            isOnHold: {
                serializedName: "isOnHold",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
export const AddParticipantRequest = {
    type: {
        name: "Composite",
        className: "AddParticipantRequest",
        modelProperties: {
            sourceCallerIdNumber: {
                serializedName: "sourceCallerIdNumber",
                type: {
                    name: "Composite",
                    className: "PhoneNumberIdentifierModel",
                },
            },
            sourceDisplayName: {
                serializedName: "sourceDisplayName",
                type: {
                    name: "String",
                },
            },
            participantToAdd: {
                serializedName: "participantToAdd",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            invitationTimeoutInSeconds: {
                constraints: {
                    InclusiveMaximum: 180,
                    InclusiveMinimum: 0,
                },
                serializedName: "invitationTimeoutInSeconds",
                type: {
                    name: "Number",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
            customCallingContext: {
                serializedName: "customCallingContext",
                type: {
                    name: "Composite",
                    className: "CustomCallingContextInternal",
                },
            },
        },
    },
};
export const AddParticipantResponse = {
    type: {
        name: "Composite",
        className: "AddParticipantResponse",
        modelProperties: {
            participant: {
                serializedName: "participant",
                type: {
                    name: "Composite",
                    className: "CallParticipantInternal",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            invitationId: {
                serializedName: "invitationId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const RemoveParticipantRequest = {
    type: {
        name: "Composite",
        className: "RemoveParticipantRequest",
        modelProperties: {
            participantToRemove: {
                serializedName: "participantToRemove",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const RemoveParticipantResponse = {
    type: {
        name: "Composite",
        className: "RemoveParticipantResponse",
        modelProperties: {
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const MuteParticipantsRequest = {
    type: {
        name: "Composite",
        className: "MuteParticipantsRequest",
        modelProperties: {
            targetParticipants: {
                serializedName: "targetParticipants",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommunicationIdentifierModel",
                        },
                    },
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const MuteParticipantsResult = {
    type: {
        name: "Composite",
        className: "MuteParticipantsResult",
        modelProperties: {
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const CancelAddParticipantRequest = {
    type: {
        name: "Composite",
        className: "CancelAddParticipantRequest",
        modelProperties: {
            invitationId: {
                serializedName: "invitationId",
                required: true,
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            operationCallbackUri: {
                serializedName: "operationCallbackUri",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const CancelAddParticipantResponse = {
    type: {
        name: "Composite",
        className: "CancelAddParticipantResponse",
        modelProperties: {
            invitationId: {
                serializedName: "invitationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const StartCallRecordingRequest = {
    type: {
        name: "Composite",
        className: "StartCallRecordingRequest",
        modelProperties: {
            callLocator: {
                serializedName: "callLocator",
                type: {
                    name: "Composite",
                    className: "CallLocator",
                },
            },
            recordingStateCallbackUri: {
                serializedName: "recordingStateCallbackUri",
                type: {
                    name: "String",
                },
            },
            recordingContentType: {
                serializedName: "recordingContentType",
                type: {
                    name: "String",
                },
            },
            recordingChannelType: {
                serializedName: "recordingChannelType",
                type: {
                    name: "String",
                },
            },
            recordingFormatType: {
                serializedName: "recordingFormatType",
                type: {
                    name: "String",
                },
            },
            audioChannelParticipantOrdering: {
                serializedName: "audioChannelParticipantOrdering",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommunicationIdentifierModel",
                        },
                    },
                },
            },
            channelAffinity: {
                serializedName: "channelAffinity",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ChannelAffinity",
                        },
                    },
                },
            },
            pauseOnStart: {
                serializedName: "pauseOnStart",
                type: {
                    name: "Boolean",
                },
            },
            recordingStorage: {
                serializedName: "externalStorage",
                type: {
                    name: "Composite",
                    className: "RecordingStorage",
                },
            },
        },
    },
};
export const ChannelAffinity = {
    type: {
        name: "Composite",
        className: "ChannelAffinity",
        modelProperties: {
            channel: {
                constraints: {
                    InclusiveMaximum: 4,
                    InclusiveMinimum: 0,
                },
                serializedName: "channel",
                type: {
                    name: "Number",
                },
            },
            participant: {
                serializedName: "participant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
        },
    },
};
export const RecordingStorage = {
    type: {
        name: "Composite",
        className: "RecordingStorage",
        modelProperties: {
            recordingStorageKind: {
                serializedName: "recordingStorageKind",
                required: true,
                type: {
                    name: "String",
                },
            },
            recordingDestinationContainerUrl: {
                serializedName: "recordingDestinationContainerUrl",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const RecordingStateResponse = {
    type: {
        name: "Composite",
        className: "RecordingStateResponse",
        modelProperties: {
            recordingId: {
                serializedName: "recordingId",
                type: {
                    name: "String",
                },
            },
            recordingState: {
                serializedName: "recordingState",
                type: {
                    name: "String",
                },
            },
            recordingKind: {
                serializedName: "recordingKind",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const AddParticipantFailed = {
    type: {
        name: "Composite",
        className: "AddParticipantFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            participant: {
                serializedName: "participant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
        },
    },
};
export const ResultInformation = {
    type: {
        name: "Composite",
        className: "ResultInformation",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "Number",
                },
            },
            subCode: {
                serializedName: "subCode",
                type: {
                    name: "Number",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const AddParticipantSucceeded = {
    type: {
        name: "Composite",
        className: "AddParticipantSucceeded",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            participant: {
                serializedName: "participant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
        },
    },
};
export const CallConnected = {
    type: {
        name: "Composite",
        className: "CallConnected",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const CallDisconnected = {
    type: {
        name: "Composite",
        className: "CallDisconnected",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const CallTransferAccepted = {
    type: {
        name: "Composite",
        className: "CallTransferAccepted",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            transferTarget: {
                serializedName: "transferTarget",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
            transferee: {
                serializedName: "transferee",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
        },
    },
};
export const CallTransferFailed = {
    type: {
        name: "Composite",
        className: "CallTransferFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const ParticipantsUpdated = {
    type: {
        name: "Composite",
        className: "ParticipantsUpdated",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            sequenceNumber: {
                serializedName: "sequenceNumber",
                type: {
                    name: "Number",
                },
            },
            participants: {
                serializedName: "participants",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CallParticipantInternal",
                        },
                    },
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const RemoveParticipantSucceeded = {
    type: {
        name: "Composite",
        className: "RemoveParticipantSucceeded",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            participant: {
                serializedName: "participant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
        },
    },
};
export const RemoveParticipantFailed = {
    type: {
        name: "Composite",
        className: "RemoveParticipantFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            participant: {
                serializedName: "participant",
                type: {
                    name: "Composite",
                    className: "CommunicationIdentifierModel",
                },
            },
        },
    },
};
export const CancelAddParticipantSucceeded = {
    type: {
        name: "Composite",
        className: "CancelAddParticipantSucceeded",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            invitationId: {
                serializedName: "invitationId",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const CancelAddParticipantFailed = {
    type: {
        name: "Composite",
        className: "CancelAddParticipantFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            invitationId: {
                serializedName: "invitationId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const AnswerFailed = {
    type: {
        name: "Composite",
        className: "AnswerFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const CreateCallFailed = {
    type: {
        name: "Composite",
        className: "CreateCallFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const RecordingStateChanged = {
    type: {
        name: "Composite",
        className: "RecordingStateChanged",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            recordingId: {
                serializedName: "recordingId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String",
                },
            },
            startDateTime: {
                serializedName: "startDateTime",
                readOnly: true,
                type: {
                    name: "DateTime",
                },
            },
            recordingKind: {
                serializedName: "recordingKind",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const PlayStarted = {
    type: {
        name: "Composite",
        className: "PlayStarted",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const PlayCompleted = {
    type: {
        name: "Composite",
        className: "PlayCompleted",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const PlayFailed = {
    type: {
        name: "Composite",
        className: "PlayFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            failedPlaySourceIndex: {
                serializedName: "failedPlaySourceIndex",
                type: {
                    name: "Number",
                },
            },
        },
    },
};
export const PlayCanceled = {
    type: {
        name: "Composite",
        className: "PlayCanceled",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const RecognizeCompleted = {
    type: {
        name: "Composite",
        className: "RecognizeCompleted",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            recognitionType: {
                serializedName: "recognitionType",
                type: {
                    name: "String",
                },
            },
            dtmfResult: {
                serializedName: "dtmfResult",
                type: {
                    name: "Composite",
                    className: "DtmfResult",
                },
            },
            choiceResult: {
                serializedName: "choiceResult",
                type: {
                    name: "Composite",
                    className: "ChoiceResult",
                },
            },
            speechResult: {
                serializedName: "speechResult",
                type: {
                    name: "Composite",
                    className: "SpeechResult",
                },
            },
        },
    },
};
export const DtmfResult = {
    type: {
        name: "Composite",
        className: "DtmfResult",
        modelProperties: {
            tones: {
                serializedName: "tones",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
export const ChoiceResult = {
    type: {
        name: "Composite",
        className: "ChoiceResult",
        modelProperties: {
            label: {
                serializedName: "label",
                type: {
                    name: "String",
                },
            },
            recognizedPhrase: {
                serializedName: "recognizedPhrase",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const SpeechResult = {
    type: {
        name: "Composite",
        className: "SpeechResult",
        modelProperties: {
            speech: {
                serializedName: "speech",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const RecognizeFailed = {
    type: {
        name: "Composite",
        className: "RecognizeFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            failedPlaySourceIndex: {
                serializedName: "failedPlaySourceIndex",
                type: {
                    name: "Number",
                },
            },
        },
    },
};
export const RecognizeCanceled = {
    type: {
        name: "Composite",
        className: "RecognizeCanceled",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const ContinuousDtmfRecognitionToneFailed = {
    type: {
        name: "Composite",
        className: "ContinuousDtmfRecognitionToneFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const ContinuousDtmfRecognitionToneReceived = {
    type: {
        name: "Composite",
        className: "ContinuousDtmfRecognitionToneReceived",
        modelProperties: {
            sequenceId: {
                serializedName: "sequenceId",
                readOnly: true,
                type: {
                    name: "Number",
                },
            },
            tone: {
                serializedName: "tone",
                type: {
                    name: "String",
                },
            },
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const ContinuousDtmfRecognitionStopped = {
    type: {
        name: "Composite",
        className: "ContinuousDtmfRecognitionStopped",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const SendDtmfTonesCompleted = {
    type: {
        name: "Composite",
        className: "SendDtmfTonesCompleted",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const SendDtmfTonesFailed = {
    type: {
        name: "Composite",
        className: "SendDtmfTonesFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const HoldFailed = {
    type: {
        name: "Composite",
        className: "HoldFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
export const ConnectFailed = {
    type: {
        name: "Composite",
        className: "ConnectFailed",
        modelProperties: {
            callConnectionId: {
                serializedName: "callConnectionId",
                type: {
                    name: "String",
                },
            },
            serverCallId: {
                serializedName: "serverCallId",
                type: {
                    name: "String",
                },
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String",
                },
            },
            operationContext: {
                serializedName: "operationContext",
                type: {
                    name: "String",
                },
            },
            resultInformation: {
                serializedName: "resultInformation",
                type: {
                    name: "Composite",
                    className: "ResultInformation",
                },
            },
        },
    },
};
